pandas/core/indexes/base.py

```diff
@@ -2211,13 +2211,32 @@ def sortlevel(self, level=None, ascending=True, sort_remaining=None):
         return self.sort_values(return_indexer=True, ascending=ascending)
 
     def shift(self, periods=1, freq=None):
-        """
-        Shift Index containing datetime objects by input number of periods and
-        DateOffset
+        """Shift index by desired number of time frequency increments.
 
-        Returns
-        -------
-        shifted : Index
+        This method is for shifting the values of datetime-like indexes
+        by a specified time increment a given number of times.
+
+        Parameters
+        ----------
+        periods : int
+            Number of periods (or increments) to shift by,
+            can be positive or negative (default is 1).
+        freq : pandas.DateOffset, pandas.Timedelta or string
+            Frequency increment to shift by (default is None).
+            Offset aliases are valid strings, e.g., 'EOM'.
+
+        See Also
+        --------
+        DatetimeIndex.shift : an implementation of shift.
+        PeriodIndex.shift : an implementation of shift.
+        TimedeltaIndex.shift : an implementation of shift.
+
+        Notes
+        -----
+        Must be overridden in child classes.
```

Comment:
This is not needed here (not relevant information for the user, but only for the developer), although could keep it as a normal comment (`# .. `) after the docstring to clarify this.

I would maybe mention in the Notes section that "This method is only implemented for DatetimeIndex, PeriodIndex and TimedeltaIndex"